CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java

```diff
@@ -2179,36 +2168,73 @@ private void updateLayerMask(final int attributeId, final GraphElementType eleme
                     }
                 }
             }
-        } else if (elementType == GraphElementType.TRANSACTION) {
-            if (transactionFilterBitmaskAttrId >= 0 && transactionLayerVisibilityAttributeId >= 0) {
-                if (attributeId != transactionFilterBitmaskAttrId) {
-                    avoidLayerUpdate = true;
-                    setIntValue(transactionFilterBitmaskAttrId, elementId, bitmask);
-                    avoidLayerUpdate = false;
+        } else if (elementType == GraphElementType.TRANSACTION && 
+                transactionFilterBitmaskAttrId != Graph.NOT_FOUND && 
+                transactionLayerVisibilityAttributeId != Graph.NOT_FOUND) {
+            if (attributeId != transactionFilterBitmaskAttrId) {
+                avoidLayerUpdate = true;
```

Comment:
The idea is that in the case of an attribute other than the Vertex/Transaction's bitmask attribute gets updated - it will not trigger another call to update the layer. `StoreGraph.setIntValue(...)` has a check just before it calls to update layers so that it avoids unnecessary recalculation. Line 2370